internal/metrics/metrics.go

```diff
@@ -1,83 +1,258 @@
 package metrics
 
 import (
+	"reflect"
+
 	"github.com/prometheus/client_golang/prometheus"
 	"sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	apis "github.com/openshift/elasticsearch-operator/apis/logging/v1"
 )
 
 const (
-	CertRestart      string = "cert_restart"
-	RollingRestart   string = "rolling_restart"
-	ScheduledRestart string = "scheduled_restart"
-	ManagedState     string = "managed"
-	UnmanagedState   string = "unmanaged"
+	labelCertRestart            string = "cert_restart"
+	labelRollingRestart         string = "rolling_restart"
+	labelScheduledRestart       string = "scheduled_restart"
+	labelManagedState           string = "managed"
+	labelUnmanagedState         string = "unmanaged"
+	labelPersistantStorage      string = "persistent"
+	labelEphemeralStorage       string = "ephemeral"
+	labelFullRedundancy         string = "full"
+	labelMultipleRedundancy     string = "multiple"
+	labelSingleRedundancy       string = "single"
+	labelZeroRedundancy         string = "zero"
+	labelRolloverIndexOperation string = "rollover"
+	labelDeleteIndexOperation   string = "delete"
 )
 
 var (
-	restartCounter = prometheus.NewCounterVec(
+	restartMetric = prometheus.NewCounterVec(
 		prometheus.CounterOpts{
-			Name: "eo_elasticsearch_cr_restart_total",
-			Help: "Total number of times the nodes restarted due to Cert Restart or Rolling Restart or Scheduled Restart.",
-		}, []string{"reason"})
+			Name: "eo_es_restart_total",
+			Help: "Number of times a node has restarted",
+		}, []string{"reason"},
+	)
 
-	esClusterManagementState = prometheus.NewGaugeVec(
+	managementStateMetric = prometheus.NewGaugeVec(
 		prometheus.GaugeOpts{
-			Name: "eo_elasticsearch_cr_cluster_management_state",
-			Help: "Number of Elasticsearch cluster that are in Managed state or Unmanaged state.",
-		}, []string{"state"})
+			Name: "eo_es_cluster_management_state_info",
+			Help: "Number of cluster that are in the managed or unmanaged",
+		}, []string{"state"},
+	)
 
-	metricList = []prometheus.Collector{
-		restartCounter,
-		esClusterManagementState,
-	}
+	storageTypeMetric = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "eo_es_storage_info",
+			Help: "Number of clusters using emphimeral or persistent storage",
+		}, []string{"type"},
+	)
+
+	redundancyPolicyTypeMetric = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "eo_es_redundancy_policy_info",
+			Help: "Number of clusters using redundancy policy",
```

Comment:
The `Help` text is not matching what the metric actually exposes.
The metric is exposing the type of redundancy policy the cluster is using.
Help text can be "Redundancy policy used by the cluster"?